clang/lib/DPCT/MigrateCmakeScript.cpp

```diff
@@ -737,18 +737,41 @@ void doCmakeScriptMigration(const clang::tooling::UnifiedPath &InRoot,
 }
 
 void registerCmakeMigrationRule(MetaRuleObject &R) {
-  auto PR =
-      MetaRuleObject::PatternRewriter(R.In, R.Out, R.Subrules, R.MatchMode,
-                                      R.RuleId, R.CmakeSyntax, R.Priority);
-
-  auto Iter = CmakeBuildInRules.find(PR.CmakeSyntax);
-  if (Iter != CmakeBuildInRules.end()) {
-    if (PR.Priority == RulePriority::Takeover) {
-      assert(Iter->second.Priority < PR.Priority &&
-             "Two same cmake syntaxes have \'Takeover\' priority.\n ");
+
+#ifdef __DEBUG_CMAKE_SCRIPR_MIGRATION
+  std::cout << "\n#####Migration Rule start ######" << std::endl;
+  std::cout << "\tRuleId:" << R.RuleId << std::endl;
+  std::cout << "\tCmakeSyntax:" << R.CmakeSyntax << std::endl;
+  std::cout << "\tIn:" << R.In << std::endl;
+  std::cout << "\tOut:" << R.Out << std::endl;
+  std::cout << "#####Migration Rule end ######" << std::endl;
+#endif
+    auto PR =
+        MetaRuleObject::PatternRewriter(R.In, R.Out, R.Subrules, R.MatchMode,
+                                        R.RuleId, R.CmakeSyntax, R.Priority);
+
+    auto Iter = CmakeBuildInRules.find(PR.CmakeSyntax);
+    if (Iter != CmakeBuildInRules.end()) {
+      if (PR.Priority == RulePriority::Takeover) {
+        if (Iter->second.Priority < PR.Priority) {
+          CmakeBuildInRules[PR.CmakeSyntax] = PR;
+        } else {
+          llvm::outs() << " [NOTE] Two Yaml rules with same CMake syntax have "
+                          "\'Takeover\' priority:\n"
+                       << "\tNew rule id:\"" << R.RuleId << "\"\n"
+                       << "\tExiting rule id:\"" << Iter->second.RuleId
+                       << "\"\n"
+                       << " The new rule has been skipped.\n";
+        }
+      } else {
```

Comment:
where is the test case for this wanning ?

Reply:
I planned to add a behavior test in the SYCLomatic test repo; while a lit test is also workable, I will update it.